examples/sht3x/sht3x_one_sensor/sht3x_one_sensor.c

```diff
@@ -0,0 +1,133 @@
+
+#ifdef SINGLE_SHOT_MODE
+/*
+ * User task that triggers measurements of sensor every 5 seconds. Due to
+ * power efficiency reasons, it uses the SHT3x *single_shot* and *vTaskDelay*
+ * to wait for measurement results.
+ */
+void user_task (void *pvParameters)
+{
+    sht3x_values_t* values;
+    int32_t duration;
+
+    while (1) 
+    {
+        // trigger one measurement in single shot mode
+        duration = sht3x_start_measurement (&sensor, single_shot);
+        
+        if (duration >= 0)
+        {
+            // passive waiting until measurement results are available
+            vTaskDelay (duration/portTICK_PERIOD_MS);
+        
+           // retrieve the values and to something with them
+           if ((values = sht3x_get_results (&sensor)))
+               printf("%.3f SHT3x Sensor: %.2f C, %.2f F, %.2f \n", 
+                      (double)sdk_system_get_time()*1e-3,  
+                      values->temperature_c, values->temperature_f, values->humidity);
+
+        }
+        // passive waiting until 5 seconds are over
+        vTaskDelay ((5000 - duration) / portTICK_PERIOD_MS);
+    }
+}
+#else  // PERIODIC MODE
+/*
+ * User task that fetches latest measurement results of sensor every 2
+ * seconds. It starts the SHT3x in periodic mode with 10 measurements per
+ * second (*periodic_10mps*). It uses busy waiting for first measurement
+ * results.
+ */
+void user_task (void *pvParameters)
+{
+    sht3x_values_t* values;
+
+    // start periodic measurement mode
+    sht3x_start_measurement (&sensor, periodic_10mps);
+
+    // busy waiting until measurement results are available
+    while (sht3x_is_measuring (&sensor) > 0) ;
+    
+    while (1) 
+    {
+        // retrieve the values and to something with them
+        if ((values = sht3x_get_results (&sensor)))
+            printf("%.3f SHT3x Sensor: %.2f C, %.2f F, %.2f \n", 
+                   (double)sdk_system_get_time()*1e-3,  
+                   values->temperature_c, values->temperature_f, values->humidity);
+
+        // passive waiting until 2 seconds are over
+        vTaskDelay (2000 / portTICK_PERIOD_MS);
+    }
+}
+#endif
+
+void user_init(void)
+{
+    // Please note: Return values are not considered in this example for 
+    // readability reasons. All functions return boolean or other values
+    // that allow effecitve error handling.
+    
+    // Set UART Parameter
+    uart_set_baud(0, 115200);
+
+    // Give the UART some time to settle
+    sdk_os_delay_us(500);
+    
+    // Init I2C bus interfaces at which SHT3x sensors are connected
+    // (different busses are possible)
+    i2c_init(I2C_BUS, I2C_SCL_PIN, I2C_SDA_PIN, I2C_FREQ_100K);
+
+    // Init SHT3x driver (parameter is the number of SHT3x sensors used)
+    sht3x_init_driver();
+    
+    // Create sensors
+    sht3x_init_sensor (&sensor, I2C_BUS, SHT3x_ADDR_1);
+    
+    // Create a task for each sensor
+    xTaskCreate(user_task, "user_task", 256, NULL, 2, 0);
+        
```

Comment:
This does not look like good advice because if there are multiple sensors then there will be multiple tasks banding on the same i2c bus without and synchronisation.

Reply:
I changed the comment so that the creation of a seperate task for each sensor is not an advice anymore. The user can decide how to deal with it.